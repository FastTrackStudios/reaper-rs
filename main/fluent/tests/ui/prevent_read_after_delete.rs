use reaper_fluent::*;

fn main() {
    let model = Reaper::get().model();
    let mut project = model.current_project();
    let track = project.tracks().next().unwrap();
    project.delete_track(track.raw());
    track.guid();
}
