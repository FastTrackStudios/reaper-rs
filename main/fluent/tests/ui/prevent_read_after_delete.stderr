error[E0502]: cannot borrow `project` as mutable because it is also borrowed as immutable
 --> tests/ui/prevent_read_after_delete.rs:7:5
  |
6 |     let track = project.tracks().next().unwrap();
  |                 ------- immutable borrow occurs here
7 |     project.delete_track(track.raw());
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
8 |     track.guid();
  |     ----- immutable borrow later used here
